runner.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+type runner struct {
+	configuration      *configuration
+	backend            backend
+	cancelChan         chan bool
+	respawnWorkerDelay time.Duration
+	respawnWorkerChan  chan *rule
+}
+
+func (rn *runner) initialize() error {
+	// Backend
```

Comment:
We should check whether configuration is nil (defensive programming)

Reply:
Revised.